syntax = "proto3";

package PGo;
import public "holoholo_shared.proto";

message FortDetailsOutProto {
  string Id = 1;
  Custom_TeamColor Team = 2;
  Custom_PokemonName Pokemon = 3;
  string Name = 4;
  string ImageUrl = 5;
  int32 Fp = 6;
  int32 Stamina = 7;
  int32 MaxStamina = 8;
  Holoholo.Rpc.FortType FortType = 9;
  double Latitude = 10;
  double Longitude = 11;
  string Description = 12;
  ClientFortModifierProto Modifier = 13;
}

message ClientFortModifierProto {
  int32 ModifierType = 1;
  int32 ExpirationTimeMs = 2;
  int32 DeployingPlayerCodename = 3;
}

message FortDetailsProto {
  string Id = 1;
  double Latitude = 2;
  double Longitude = 3;
}

message FortSearchProto {
  string Id = 1;
  double PlayerLatDegrees = 2;
  double PlayerLngDegrees = 3;
  double FortLatDegrees = 4;
  double FortLngDegrees = 5;
}

message FortSearchOutProto {
  int32 Result = 1;
  repeated ItemProto Items = 2;
  int32 GemsAwarded = 3;
  int32 EggPokemon = 4;
  int32 XpAwarded = 5;
  uint64 CooldownComplete = 6;
  int32 ChainHackSequenceNumber = 7;
}

message ItemProto {
  Holoholo.Rpc.Item Item = 1;
  int32 Count = 2;
  bool Unseen = 3;
}


message Gym {
  message Details {
    message MoreDetails {
      string id = 1;
      uint64 timestamp = 2;
      double lat = 3;
      double long = 4;
      //5, 6, 8
      uint64 prestige = 10;
    }
    message Level {
      message Trainer {
        string name = 1;
        uint64 level = 2;
      }

      PokemonProto pokemon = 1;
      Trainer trainer = 2;
    }

    MoreDetails details = 1;
    repeated Level levels = 2;
  }

  Details details = 1;
  string name = 2;
  string image_url = 3;
  uint64 team = 4;
  string description = 5;
}

message GetGymDetailsProto {
  string GymId = 1;
  double PlayerLatDegrees = 2;
  double PlayerLngDegrees = 3;
  double GymLatDegrees = 4;
  double GymLngDegrees = 5;
}

message GetGymDetailsOutProto {
  enum ResultEnum {
    UNSET = 0;
    SUCCESS = 1;
    ERROR_NOT_IN_RANGE = 2;
  }

  GymStateProto GymState = 1;
  string Name = 2;
  string Url = 3;
  ResultEnum Result = 4;
  string Description = 5;
}

message GymStateProto {
  PokemonFortProto FortMapData = 1;
  //unknown GymMembership = 2;
}

